app/services/one_off/fhir/appointment_exporter.rb

```diff
@@ -0,0 +1,57 @@
+require "fhir_models"
+
+module OneOff
+  module Fhir
+    class AppointmentExporter
+      attr_reader :appointment
+
+      def initialize(appointment)
+        @appointment = appointment
+      end
+
+      def export
+        FHIR::Appointment.new(
+          status: appointment_status_code,
+          start: appointment.scheduled_date.beginning_of_day.iso8601,
+          cancellationReason: cancellation_code,
+          participant: [
+            FHIR::Appointment::Participant.new(
+              actor: FHIR::Reference.new(
+                id: FHIR::Patient.new(
+                  identifier: FHIR::Identifier.new(
+                    value: appointment.patient_id
+                  )
+                )
+              )
```

Comment:
[question] Can you verify if this is the correct way to build a reference?

The library is unfortunately sparsely documented, but the source code suggests that a hash is expected here, not a reference to another model?
https://github.com/fhir-crucible/fhir_models/blob/be9087e8fc0e49dc93f504d18ceb76cbb32b3811/lib/fhir_models/bootstrap/model.rb#L9-L18

Reply:
This comment applies to every reference in this patch.

Reply:
Thinking I'll go ahead with [this](https://github.com/simpledotorg/simple-server/pull/5335#discussion_r1432810127) approach and just pass whatever resource id on the `identifier` key of a Reference. No validation errors.

Reply:
Don't know if I followed the question, you're saying the actor field expects a hash?

But re: building references correctly:

Using id was a hack to get the validation checks to pass a while ago. From [this](https://github.com/simpledotorg/simple-server/pull/5335#discussion_r1432810127), I'm going to rewrite all the references to directly use the `identifier` key and pass to it the respective id. In this example, rewrite this as:
```
 actor: FHIR::Reference.new(
    identifier: FHIR::Identifier.new(
      value: appointment.patient_id
   )
 )
```
No validation errors, and returns the expected json:
```
{... {"actor"=>{"identifier"=>{"value"=>"1dae7c00-7e55-4879-be50-184b2f535e0f"}}} ... }
```

Reply:
Lovely. This explanation unconfused me.